### PR TITLE
Fixed fishing rod duping with poly belts and shapeshift spells.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
It turns out the "shapeshifted from spell" status ejects everything inside the shapeshifted mob when removed. That's been causing a little issue with the fishing rod from the profound_fisher component, which a few mobs have. This PR fixes just that.

## Why It's Good For The Game
immagine
immagine

## Changelog

:cl:
fix: Fixed fishing rod duping with poly belts and shapeshift spells.
spellcheck: Fixed a small typo when examining fishing rods.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
Two things:
- Why isn't the rod deleted on component destruction anyway? It should be removed when the component is removed, which would happen when the mob is deleted. 
- What about some handling in the component for if the fishing rod leaves mob contents in order to avoid this reoccuring in future via some other means? Not sure if there is a signal on move that can be blocked, but you could either move it back, destroy the component, or destroy and recreate the rod.

**Comment by Ghommie** [comment]
Good points, I'll do something about it.

**Comment by Ghommie** [comment]
Done.

File: code/datums/components/profound_fisher.dm [hunk 1]
```diff
@@ -2,15 +2,18 @@
 /datum/component/profound_fisher
 	///the fishing rod this mob will use
 	var/obj/item/fishing_rod/mob_fisher/our_rod
+	///Wether we should delete the fishing rod along with the component or replace it if it's somehow removed from the parent
+	var/delete_rod = TRUE
```
Comment (Jacquerel) [inline code/datums/components/profound_fisher.dm:6]: I think the variable name could be a little more descriptive about what circumstances under which we delete the rod, the comment elaborates but self-describing is better.

File: code/datums/components/profound_fisher.dm [hunk 2]
```diff
@@ -37,14 +40,26 @@
...
 /datum/component/profound_fisher/Destroy()
-	our_rod.internal = FALSE
-	UnregisterSignal(our_rod, COMSIG_QDELETING)
-	REMOVE_TRAIT(our_rod, TRAIT_NOT_BARFABLE, REF(src))
+	UnregisterSignal(our_rod, COMSIG_MOVABLE_MOVED)
+	if(!delete_rod)
+		our_rod.internal = FALSE
+		REMOVE_TRAIT(our_rod, TRAIT_NOT_BARFABLE, REF(src))
+	else if(QDELETED(our_rod))
+		QDEL_NULL(our_rod)
```
Comment (Jacquerel) [inline code/datums/components/profound_fisher.dm:62]: Not sure this is right?
If the rod is already being deleted you don't need to delete it again, just set it to null (which you will do anyway). If this code was instead _intended_ to delete the rod on component removal, I think it doesn't do that.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/datums/components/profound_fisher.dm (hunk 2 above)
Comment (Ghommie) [reply code/datums/components/profound_fisher.dm:62]: Oh, good catch

**Pull request merged**